### PR TITLE
Updates for loadByPrefix public autoloader

### PR DESCRIPTION
**Pull request by ghost**
In testing, I got some odd errors.  The spl_auto_register call would throw exceptions when trying to register _autoload because _autoload is not a public method.

Rather than make a method with an underscore public[since by convention it's private] - I instead duplicated it to loadByPrefix.  It is more clear any way since it matches loadByNamespace....

### CODE REVIEW

**Comment by ghost** [comment]
Modified pull request to remove changes to unrelated JController file made for testing.

File: tests/suites/unit/JLoaderTest.php [hunk 1]
```diff
@@ -577,6 +577,7 @@ public function testSetupDefaultParameters()
 		$foundLoadByNamespaceLowerCase = false;
 		$loadByNamespaceNaturalCase = false;
 		$loadByNamespaceMixedCase = false;
+        $loadByPrefix = false;
```
Comment (elinw) [inline tests/suites/unit/JLoaderTest.php]: Is there a reason for the extra space here?

File: tests/suites/unit/JLoaderTest.php [hunk 2]
```diff
@@ -593,6 +594,10 @@ public function testSetupDefaultParameters()
 					$foundAutoload = true;
 				}
 
+                if ($loader[1] === 'loadByPrefix')
+                {
+                    $loadByPrefix = true;
+                }
```
Comment (elinw) [inline tests/suites/unit/JLoaderTest.php]: I think this has extra spacing, also remember to add a space before the following if.

File: tests/suites/unit/JLoaderTest.php [hunk 3]
```diff
@@ -617,8 +622,11 @@ public function testSetupDefaultParameters()
 		$prefixes = TestReflection::getValue('JLoader', 'prefixes');
 		$this->assertArrayHasKey('J', $prefixes);
 
-		// Assert the prefix loader is found.
-		$this->assertTrue($foundAutoload);
+		// Assert the default autoloader is not set
+		$this->assertFalse($foundAutoload);
+
+        // Assert the prefix loader is found.
+        $this->assertTrue($loadByPrefix);
```
Comment (elinw) [inline tests/suites/unit/JLoaderTest.php]: Why the indenting here?

File: tests/suites/unit/JLoaderTest.php [hunk 4]
```diff
@@ -696,12 +704,19 @@ public function testSetupWithoutPrefixes()
...
+                }
 			}
 		}
 
 		// We don't expect to find it.
 		$this->assertFalse($foundAutoLoad);
 
+        // in either format
+        $this->assertTrue($loadByPrefix);
+
```
Comment (elinw) [inline tests/suites/unit/JLoaderTest.php:719]: Capitalize the first word in an inline comment please.

Why indented?

File: tests/suites/unit/JLoaderTest.php (hunk 1 above)
Comment (ghost) [inline tests/suites/unit/JLoaderTest.php]: Not showing up in my editor...not sure if there is a tab somewhere[I have show hidden chars on..]

**Comment by ghost** [comment]
Hmm, I think thats all of them.

**Comment by eddieajau** [comment]
@garyamort is this PR still useful for the CMS do you think?

**Comment by dongilbert** [comment]
@garyamort There are 14 checkstyle issues with using spaces to indent instead of tabs, etc. http://developer.joomla.org/pulls/pulls/1707.html

Can you fix them up and leave a comment saying "reopen".

**Pull request closed without merging**